cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricsWriterContext.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.metrics;
+
+/**
+ * Context passed to {@link MetricsWriter#initialize(MetricsContext, MetricsWriterContext)}.
+ */
+public final class MetricsWriterContext {
+  private final String metricsWriterFilePath;
+  private final MetricsContext metricsContext;
+
+  public MetricsWriterContext(MetricsContext metricsContext, String metricsWriterFilePath) {
+    this.metricsWriterFilePath = metricsWriterFilePath;
+    this.metricsContext = metricsContext;
+  }
+
+  public  String getMetricsWriterFilePath() {
```

Comment:
Remove extra space. Also, since this is an API object, please add javadoc for all methods. Also, it is better for this to be an interface and have the implementation separated out in the non-api module.

Reply:
and why is a file path needs to be available at the context object? Is it generic?

Reply:
> and why is a file path needs to be available at the context object?  Is it generic?

The file path is only used by the stackdriver extension, not used by `MetricStoreMetricsWriter`. If we remove the file path from the context object, how will the stackdriver extension get the file path?

Reply:
Fixed.